handlers/canvas_handler.py

```diff
@@ -168,6 +206,33 @@ def untrack_course(self, course_ids_str: Tuple[str, ...]):
             if c in self.due_day:
                 del self.due_day[c]
 
+        for i in ids_of_removed_courses:
+            watchers_file = f'{COURSES_DIRECTORY}/{i}/watchers.txt'
+            self.delete_channels_from_file(self.live_channels, watchers_file)
+
+            # If there are no more channels watching the course, we should delete that course's directory.
+            if os.stat(watchers_file).st_size == 0:
+                shutil.rmtree(f'{COURSES_DIRECTORY}/{i}')
+        
+    @staticmethod
+    def delete_channels_from_file(text_channels: List[discord.TextChannel], file_path: str):
+        """
+        For each text channel provided, we remove its id from the file with given path
+        if the id is contained in the file.
+        """
+
+        util.create_file_if_not_exists(file_path)
+
+        with open(file_path, "r") as f:
+            channel_ids = f.readlines()
+
+        ids_to_remove = set(map(lambda channel: str(channel.id) + "\n", text_channels))
+
+        with open(file_path, "w") as f:
+            for channel_id in channel_ids:
+                if not channel_id in ids_to_remove:
```

Comment:
Generally, this would be written with not in though it is personal style. This could also be done with filterfalse.